packages/adapter-mongoose/src/mongoose.ts

```diff
@@ -83,6 +83,43 @@ export const mongooseAdapter = (models: {
 				).lean();
 				return sessions.map((val) => transformSessionDoc(val));
 			},
+			getSessionAndUserBySessionId: async (sessionId: string) => {
+				if (!Session) {
+					throw new Error("Session model not defined");
+				}
+
+				const sessionUser = await Session.aggregate([
+					{ $match: { _id: sessionId } },
+					{
+						$lookup: {
+							from: User.collection.name,
+							localField: "user_id",
+							// Relies on _id being a String, not ObjectId. 
+							// But this assumption is used elsewhere, as well
+							foreignField: "_id",
+							as: "users"
+						}
+					}
+				]).exec();
+
+				// More verbose, if you prefer:
```

Comment:
Can you change this part to something like this? Prefer if we don't use optional chaining inside `if` checks.

```ts
const sessionUsers = await Session.aggregate({
  // ...
});

const sessionUser = sessionUsers?.at(0) ?? null;

if (!sessionUser) return null
const { users: userDocs, ...sessionDoc } = sessionUser;
const userDoc = userDocs?.at(0) ?? null;

if (!userDoc) return null;

// ...
```

Reply:
Yes of course. Please confirm I've done the auri change correctly.

Reply:
I've also changed the checks as you suggested